### PR TITLE
Fix buildDependencyGraph implementation to loop over builders under

### PR DESCRIPTION
**Pull request by williambernardet**
Looks like I did a nasty copy/paste mistake in my previous commit.
ProxyBuilder should iterate over builders inside the buildDependencyGraph method, rather than publisher.

### CODE REVIEW

File: src/main/java/hudson/plugins/templateproject/ProxyBuilder.java [hunk 1]
```diff
@@ -51,11 +54,12 @@ public String getProjectName() {
 
 	@Override
 	public void buildDependencyGraph(AbstractProject project, DependencyGraph graph) {
-		AbstractProject<?, ?> templateProject = (AbstractProject) Hudson.getInstance().getItem(getProjectName());
-		if (templateProject != null) {
-			for (Publisher publisher : templateProject.getPublishersList().toList()) {
-				if (publisher instanceof DependecyDeclarer) {
-					((DependecyDeclarer)publisher).buildDependencyGraph(project, graph);
+		TopLevelItem item = Jenkins.getInstance().getItem(getProjectName());
```
Comment (Brantone) [inline src/main/java/hudson/plugins/templateproject/ProxyBuilder.java]: This should probably be `getItemByFullName` to account for Folder plugin, shouldn't it?

Comment (williambernardet) [inline src/main/java/hudson/plugins/templateproject/ProxyBuilder.java]: Probably and to be consistent then it should be updated under getProjectBuilders method as well.

**Comment by Brantone** [comment]
Bunch of excessive commits/merges ... possible to do a rebase to clean it up a bit?

**Comment by williambernardet** [comment]
Finally managed to merge it to one commit...

**Pull request merged**